packages/lib-classifier/src/plugins/drawingTools/components/draggable/draggable.js

```diff
@@ -1,137 +1,118 @@
-  class Draggable extends PureComponent {
-    constructor(props) {
-      super(props)
-      this.dragStart = this.dragStart.bind(this)
-      this.dragMove = this.dragMove.bind(this)
-      this.dragEnd = this.dragEnd.bind(this)
-      this.wrappedComponent = createRef()
-      this.state = {
-        coords: {
-          x: props.coords.x,
-          y: props.coords.y
-        },
-        dragging: false,
-        pointerId: -1
-      }
-    }
-
-    convertEvent(event) {
-      const type = event.type
-
-      const svgEventOffset = this.getEventOffset(event)
+function createPoint(event) {
+  const { clientX, clientY } = event
+  // SVG 2 uses DOMPoint
+  if (window.DOMPointReadOnly) {
+    return new DOMPointReadOnly(clientX, clientY)
+  }
+  // jsdom doesn't support SVG
+  return {
+    x: clientX,
+    y: clientY
+  }
+}
 
-      const svgCoordinateEvent = {
-        pointerId: event.pointerId,
-        type,
-        x: svgEventOffset.x,
-        y: svgEventOffset.y
-      }
+function getEventOffset(event, canvas) {
+  const { clientX, clientY } = event
+  const svgPoint = createPoint(event)
+  const svgEventOffset = svgPoint.matrixTransform
+    ? svgPoint.matrixTransform(canvas.getScreenCTM().inverse())
+    : svgPoint
+  return svgEventOffset
+}
 
-      return svgCoordinateEvent
-    }
+function convertEvent(event, canvas) {
+  const svgEventOffset = getEventOffset(event, canvas)
+  const svgCoordinateEvent = {
+    pointerId: event.pointerId,
+    type: event.type,
+    x: svgEventOffset.x,
+    y: svgEventOffset.y
+  }
 
-    getBoundingClientRect() {
-      return this.wrappedComponent.current.getBoundingClientRect()
-    }
+  return svgCoordinateEvent
+}
 
-    createPoint(event) {
-      const { clientX, clientY } = event
-      // SVG 2 uses DOMPoint
-      if (window.DOMPointReadOnly) {
-        return new DOMPointReadOnly(clientX, clientY)
-      }
-      // jsdom doesn't support SVG
-      return {
-        x: clientX,
-        y: clientY
-      }
-    }
+const DEFAULT_HANDLER = event => true
 
-    getEventOffset(event) {
-      const { clientX, clientY } = event
-      const { canvas } = this.context
-      const svgPoint = this.createPoint(event)
-      const svgEventOffset = svgPoint.matrixTransform
-        ? svgPoint.matrixTransform(canvas.getScreenCTM().inverse())
-        : svgPoint
-      return svgEventOffset
+function draggable(WrappedComponent) {
+  function Draggable({
+    children,
+    coords: initialCoords,
+    dragStart = DEFAULT_HANDLER,
+    dragMove = DEFAULT_HANDLER,
+    dragEnd = DEFAULT_HANDLER,
+    ...rest
+  }) {
+    const { canvas } = useContext(SVGContext)
+    const wrappedComponent = useRef()
+    const [dragging, setDragging] = useState(false)
+    const [coords, setCoords] = useState(initialCoords)
+    const [pointerId, setPointerId] = useState(-1)
+
+    function getBoundingClientRect() {
+      return wrappedComponent.current.getBoundingClientRect()
     }
```

Comment:
This was an instance method, used to calculate the bounds of drawn marks. 

```js
mark.getBoundingClientRect()
```

I think it can be deleted, since functional components don't have instance methods.